test/cql-pytest/test_describe.py

```diff
@@ -1372,52 +1373,54 @@ def get_create_stmt(self) -> str:
 
 ###
 
-def test_create_role_with_salted_hash(cql):
+def test_create_role_with_hashed_password(cql):
     """
-    Verify that creating a role with a salted hash works correctly, i.e. that the salted hash
+    Verify that creating a role with a hashed password works correctly, i.e. that the hashed password
     present in `system.roles` is the same as the one we provide.
     """
 
     with AuthSLContext(cql):
         role = "andrew"
-        # Arbitrary salted hash. Could be anything.
+        # Arbitrary hashed password. Could be anything.
         # We don't use characters that won't be generated, i.e.:
         #    `:`, `;`, `*`, `!`, and `\`,
         # but Scylla should technically accept them too.
-        salted_hash = "@#$%^&()`,./{}[]abcdefghijklmnopqrstuwvxyzABCDEFGHIJKLMNOPQRSTUWVXYZ123456789~-_=+|"
-        cql.execute(f"CREATE ROLE {role} WITH SALTED HASH = '{sanitize_password(salted_hash)}'")
+        hashed_password = "$2a$10$JSJEMFm6GeaW9XxT5JIheuEtPvat6i7uKbnTcxX3c1wshIIsGyUtG"
+        cql.execute(f"CREATE ROLE {role} WITH HASHED PASSWORD = '{sanitize_password(hashed_password)}'")
 
-        [result] = cql.execute(f"SELECT salted_hash FROM system.roles WHERE role = '{role}'")
-        assert salted_hash == result.salted_hash
+        table = "system.roles" if is_scylla(cql) else "system_auth.roles"
+        [result] = cql.execute(f"SELECT salted_hash FROM {table} WHERE role = '{role}'")
+        assert hashed_password == result.salted_hash
 
 
-def test_create_role_with_salted_hash_authorization(cql):
+def test_create_role_with_hashed_password_authorization(cql):
     """
-    Verify that roles that aren't superusers cannot perform `CREATE ROLE WITH SALTED HASH`.
+    Verify that roles that aren't superusers cannot perform `CREATE ROLE WITH HASHED PASSWORD`.
     """
 
     with AuthSLContext(cql):
-        def try_create_role_with_salted_hash(role):
+        def try_create_role_with_hashed_password(role):
             with new_session(cql, role) as ncql:
                 with pytest.raises(Unauthorized):
-                    ncql.execute("CREATE ROLE some_unused_name WITH SALTED HASH = 'somesaltedhash'")
+                    ncql.execute("CREATE ROLE some_unused_name WITH HASHED PASSWORD = '$2a$10$JSJEMFm6GeaW9XxT5JIheuEtPvat6i7uKbnTcxX3c1wshIIsGyUtG'")
 
         # List of form (role name, list of permission grants to the role)
         r1 = "andrew"
         r2 = "jane"
+        r3 = "bob"
 
-        with new_user(cql, r1), new_user(cql, r2):
-            # This also grants access to system tables.
-            cql.execute(f"GRANT ALL ON ALL KEYSPACES TO {r2}")
+        for r in [r1, r2]:
+            cql.execute(f"CREATE ROLE {r} WITH PASSWORD = '{r}' AND LOGIN = true")
+        cql.execute(f"CREATE ROLE {r3} WITH PASSWORD = '{r3}' AND SUPERUSER = true AND LOGIN = true")
```

Comment:
So it looks like this test was never correct, it never had one user with userperuser and one without? :-(
By the way, if I remember correctly, new_user() has login and superuser options (right?) so you could have used them.

Reply:
The test was correct, at least as far as I can tell, but for some reason didn't work correctly against Cassandra: it reported that the user couldn't log in even though `new_user` sets `LOGIN = true` by default (in fact, it doesn't give us any other choice there). But it's highly likely that I made a mistake somewhere. There shouldn't be any semantic difference between these two versions, so it should be fine